tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -84,37 +84,42 @@ ROOT::Experimental::RResult<std::uint64_t> ROOT::Experimental::Internal::RPageSt
 
 //------------------------------------------------------------------------------
 
-void ROOT::Experimental::Internal::RPageSource::RActivePhysicalColumns::Insert(DescriptorId_t physicalColumnID)
+void ROOT::Experimental::Internal::RPageSource::RActivePhysicalColumns::Insert(
+   DescriptorId_t physicalColumnId, RColumnElementBase::RIdentifier elementId)
 {
-   for (unsigned i = 0; i < fIDs.size(); ++i) {
-      if (fIDs[i] == physicalColumnID) {
-         fRefCounters[i]++;
+   auto [itr, _] = fColumnInfos.emplace(physicalColumnId, std::vector<RColumnInfo>());
+   for (auto &columnInfo : itr->second) {
+      if (columnInfo.fElementId == elementId) {
+         columnInfo.fRefCounter++;
          return;
       }
    }
-   fIDs.emplace_back(physicalColumnID);
-   fRefCounters.emplace_back(1);
+   itr->second.emplace_back(RColumnInfo{elementId, 1});
 }
 
-void ROOT::Experimental::Internal::RPageSource::RActivePhysicalColumns::Erase(DescriptorId_t physicalColumnID)
+void ROOT::Experimental::Internal::RPageSource::RActivePhysicalColumns::Erase(DescriptorId_t physicalColumnId,
+                                                                              RColumnElementBase::RIdentifier elementId)
 {
-   for (unsigned i = 0; i < fIDs.size(); ++i) {
-      if (fIDs[i] == physicalColumnID) {
-         if (--fRefCounters[i] == 0) {
-            fIDs.erase(fIDs.begin() + i);
-            fRefCounters.erase(fRefCounters.begin() + i);
-         }
-         return;
+   auto itr = fColumnInfos.find(physicalColumnId);
+   R__ASSERT(itr != fColumnInfos.end());
+   for (std::size_t i = 0; i < itr->second.size(); ++i) {
+      if (itr->second[i].fElementId != elementId)
+         continue;
+
+      itr->second[i].fRefCounter--;
+      if (itr->second[i].fRefCounter == 0) {
+         itr->second.erase(itr->second.begin() + i);
```

Comment:
This may erase the last requested in-memory representation, resulting in `itr->second.empty()`. Do we then want to `fColumnInfos.erase(itr)`?